From 71b84fca5e2af0b73e7eff2ea017b246b273dc8b Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Wed, 4 Sep 2013 09:41:13 +0800
Subject: [PATCH 2/5] ENGR00275681 Camera application exited when take picture

When sleep, OnPause will release mCameraSound,
when wakeup, doOnResume should make sure mCameraSound is created

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 src/com/android/camera/Camera.java | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/com/android/camera/Camera.java b/src/com/android/camera/Camera.java
index 2ca7944..b7cfe9f 100644
--- a/src/com/android/camera/Camera.java
+++ b/src/com/android/camera/Camera.java
@@ -1570,6 +1570,8 @@ public class Camera extends ActivityBase implements FocusManager.Listener,
         }
         // Dismiss open menu if exists.
         PopupManager.getInstance(this).notifyShowPopup(null);
+
+	if (mCameraSound == null) mCameraSound = new MediaActionSound();
     }
 
     @Override
@@ -1578,7 +1580,10 @@ public class Camera extends ActivityBase implements FocusManager.Listener,
         stopPreview();
         // Close the camera now because other activities may need to use it.
         closeCamera();
-        if (mCameraSound != null) mCameraSound.release();
+        if (mCameraSound != null) {
+			mCameraSound.release();
+			mCameraSound = null;
+        }
         resetScreenOn();
 
         // Clear UI.
-- 
1.8.0

